test/webpack-s3-uploader.test.js

```diff
@@ -30,88 +17,38 @@ describe('--- WebpackS3Uploader ---', () => {
 
   describe('constructor', () => {
 
-    it('throws an error when missing `whitelist` options', () => {
-      expect(() => new WebpackS3Uploader()).to.throw('WebpackS3Uploader: `whitelist` is a required option');
-    });
-
     it('throws an error when missing `directory` options', () => {
       expect(() => new WebpackS3Uploader({
-        whitelist: ['js']
       })).to.throw('WebpackS3Uploader: `directory` is a required option');
     });
 
     it('set\'s up default options', () => {
-      const plugin = new WebpackS3Uploader(defaultOptions);
+      const plugin = new WebpackS3Uploader(options);
       expect(plugin.options.logger).to.be.undefined;
-      expect(plugin.options.basePath).to.equal('');
+      expect(plugin.options.basePath).to.be.equal('');
+      expect(plugin.options.directory).to.be.equal(options.directory);
+      expect(plugin.options.whitelist).to.be.deep.equal(['js', 'css', 'map']);
+      expect(plugin.options.s3Options).to.be.empty;
+      expect(plugin.options.s3UploadOptions).to.be.empty;
     });
 
   });
 
   describe('apply', () => {
-
     it('adds a callback to the after-emit event', () => {
       const plugin = new WebpackS3Uploader(options);
       const compilerMock = {
-        plugin: sandbox.spy()
-      };
-
-      plugin.apply(compilerMock);
-
-      expect(compilerMock.plugin.called).to.be.true;
-      expect(compilerMock.plugin.firstCall.args[0]).to.equal('after-emit');
-    });
-
-    it('outputs success message after upload is finished', (done) => {
```

Comment:
A couple of tests removed here. Are we still covering the functionality with other tests?

Reply:
I needed some feedback on this actually, since the API is changed now, and it gets a function as a parameter, I didn't find a good way to test it. Any thoughts?

Reply:
Hmm, i'll take a look 👍